ui-admin/src/study/surveys/SurveyEnvironmentTable.tsx

```diff
@@ -31,52 +35,95 @@ type SurveyEnvTableRow = {
+  const rowInfo: SurveyEnvTableRow[] = useMemo(() => orderedStableIds.map(stableId => {
     const info: SurveyEnvTableRow = {
-      stableId,
+      stableId: stableId as string,
       name: configuredSurveys.find(config => config.survey.stableId === stableId)?.survey?.name ?? ''
     }
     ENVIRONMENT_NAMES.forEach(envName => {
-      info[envName] = configuredSurveys.find(config =>
-        config.survey.stableId === stableId && config.envName === envName)
+      info[envName] = configForEnv(stableId as string, envName, configuredSurveys)
     })
     return info
   }), [dependencyString])
 
-  const columns: ColumnDef<SurveyEnvTableRow>[] = useMemo(() => [{
-    header: '',
-    id: 'name',
-    cell: ({ row }) => <Link to={`surveys/${row.original.stableId}`}>
-      {row.original.name}
-    </Link>
-  },
-  ...ENVIRONMENT_NAMES.map((envName): ColumnDef<SurveyEnvTableRow> => ({
-    header: envName,
-    enableSorting: false,
-    id: envName,
-    cell: ({ row }) => <SurveyTableEnvColumn rowInfo={row.original} {...props} envName={envName}/>
-  }))], [dependencyString])
+  const columns: ColumnDef<SurveyEnvTableRow>[] = useMemo(() => {
+    const baseCols = [{
+      header: '',
+      id: 'name',
+      cell: ({ row }) => <div className="h-100 align-items-center d-flex">
+        <Link to={`surveys/${row.original.stableId}`}>
+          {row.original.name}
+        </Link>
+      </div>
+    },
+    ...ENVIRONMENT_NAMES.map((envName): ColumnDef<SurveyEnvTableRow> => ({
+      header: envName,
+      enableSorting: false,
+      id: envName,
+      cell: ({ row }) => <SurveyTableEnvColumn rowInfo={row.original} {...props} envName={envName}/>
+    }))]
+    // add drag handle column if in sandbox
+    if (props.studyEnvParams.envName === 'sandbox') {
+      baseCols.unshift({
+        header: '',
+        id: 'drag-handle',
+        cell: ({ row }) => <RowDragHandleCell rowId={row.id} />,
+        size: 30
+      })
+    }
+    return baseCols
+  }, [])
 
   const table = useReactTable({
     data: rowInfo,
     columns,
+    getRowId: row => row.stableId, //required because row indexes will change
     enableRowSelection: true,
     getCoreRowModel: getCoreRowModel(),
     getSortedRowModel: getSortedRowModel()
   })
 
+  const saveOrdering = () => {
+    const studyEnvSurveys: StudyEnvironmentSurvey[] =
+      rowInfo.filter(row => row.sandbox)
+        .map((row, index) => ({
+          ...row.sandbox as StudyEnvironmentSurvey,
+          surveyOrder: index
+        }))
+    props.updateConfiguredSurveys(studyEnvSurveys)
+  }
+
+  const isReordered = dependencyString !== stableIds.join('^')
+  const renderedTable = useDraggableTableLayout(table, {},
+    orderedStableIds, setOrderedStableIds, (row: Row<SurveyEnvTableRow>) => row.original.stableId)
+
   return <div>
-    { basicTableLayout(table) }
+    { renderedTable }
+    { isReordered && <div>
+      <Button variant="secondary" onClick={saveOrdering}>Save new order</Button>
```

Comment:
good suggestions.  I like the idea of it always visible, and now it becomes primary when the table is rerodered as a hint 'save now!'  

<img width="438" alt="image" src="https://github.com/user-attachments/assets/74c34e74-f04b-4f8f-8ef1-ba9108e9e075">

<img width="388" alt="image" src="https://github.com/user-attachments/assets/72ac4ad3-f1fa-4824-bf0a-90d7abd7ddee">

I kept it at the bottom because that's why I look for save buttons when there are multiple things on the page and that's where the 'add' was